capsules/src/lsm303dlhc.rs

```diff
@@ -0,0 +1,801 @@
+                    true
+                } else {
+                    self.temperature_client.map(|client| {
+                        client.callback(0);
+                    });
+                    false
+                };
+                if values {
+                    self.callback.map(|callback| {
+                        callback.schedule(temp, 0, 0);
+                    });
+                } else {
+                    self.callback.map(|callback| {
+                        callback.schedule(0, 0, 0);
+                    });
+                }
+                self.buffer.replace(buffer);
+                self.state.set(State::Idle);
+            }
+            State::ReadMagnetometerXYZ => {
+                let mut x: usize = 0;
+                let mut y: usize = 0;
+                let mut z: usize = 0;
+                let values = if error == Error::CommandComplete {
+                    self.nine_dof_client.map(|client| {
+                        // compute using only integers
+                        let range = self.mag_range.get() as usize;
+                        x = (((buffer[1] as i16 | ((buffer[0] as i16) << 8)) as i32) * 100
+                            / RANGE_FACTOR_X_Y[range] as i32) as usize;
+                        z = (((buffer[3] as i16 | ((buffer[2] as i16) << 8)) as i32) * 100
+                            / RANGE_FACTOR_X_Y[range] as i32) as usize;
+                        y = (((buffer[5] as i16 | ((buffer[4] as i16) << 8)) as i32) * 100
+                            / RANGE_FACTOR_Z[range] as i32) as usize;
+                        client.callback(x, y, z);
+                    });
+
+                    x = ((buffer[1] as u16 | ((buffer[0] as u16) << 8)) as i16) as usize;
+                    z = ((buffer[3] as u16 | ((buffer[2] as u16) << 8)) as i16) as usize;
+                    y = ((buffer[5] as u16 | ((buffer[4] as u16) << 8)) as i16) as usize;
+                    true
+                } else {
+                    self.nine_dof_client.map(|client| {
+                        client.callback(0, 0, 0);
+                    });
+                    false
+                };
+                if values {
+                    self.callback.map(|callback| {
+                        callback.schedule(x, y, z);
+                    });
+                } else {
+                    self.callback.map(|callback| {
+                        callback.schedule(0, 0, 0);
+                    });
+                }
+                self.buffer.replace(buffer);
+                self.state.set(State::Idle);
+            }
+            _ => {
+                debug!("buffer {:?} error {:?}", buffer, error);
+                self.buffer.replace(buffer);
+            }
+        }
+    }
+}
+
+impl Driver for Lsm303dlhcI2C<'a> {
+    fn command(&self, command_num: usize, data1: usize, data2: usize, _: AppId) -> ReturnCode {
+        match command_num {
+            0 /* check if present */ => ReturnCode::SUCCESS,
```

Comment:
```suggestion
            0 => ReturnCode::SUCCESS,
```
Helps with rustfmt issue.